tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/ZipUtil.java

```diff
@@ -0,0 +1,66 @@
+package com.linagora.tmail.james.jmap;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.james.jmap.routes.Blob;
+
+import reactor.core.publisher.Flux;
+
+public class ZipUtil {
+    public static Flux<byte[]> createZipStream(Collection<Blob> blobs) {
+        return Flux.create(sink -> {
+            try (ZipOutputStream zos = new ZipOutputStream(new OutputStream() {
+                @Override
+                public void write(int b) {
+                    sink.next(new byte[]{(byte) b});
+                }
+
+                @Override
+                public void write(byte[] b, int off, int len) {
+                    sink.next(java.util.Arrays.copyOfRange(b, off, off + len));
```

Comment:
We shall rather use ByteBuf `Unpooled.wrappedBuffer(b, off, len)` which would actually prevent a copy (so the sink shall be a `Sink<ByteBuf>`).

Reply:
MMMMMM not applied